### PR TITLE
MGMT-10125: Enhance logging for package storage

### PR DESCRIPTION
**Pull request by yevgeny-shnaidman**

### CODE REVIEW

File: pkg/storage/storage.go [hunk 1]
```diff
@@ -62,16 +60,14 @@ func (s *storage) UpdateConfigMapEntry(ctx context.Context, key string, value st
 func (s *storage) DeleteConfigMapEntry(ctx context.Context, key string, ins types.NamespacedName) error {
 	cm, err := s.getConfigMap(ctx, ins.Namespace, ins.Name)
 	if err != nil {
-		ctrl.LoggerFrom(ctx).Error(err, "Failed to get configmap to remove an entry", "namespacedName", ins, "key", key)
-		return err
+		return fmt.Errorf("failed to get configmap %s/%s: %w", ins.Namespace, ins.Name, err)
```
[the remark this answers is not in the document]
Comment (yevgeny-shnaidman) [reply pkg/storage/storage.go]: fixed

File: pkg/storage/storage.go [hunk 2]
```diff
@@ -51,8 +50,7 @@ func (s *storage) UpdateConfigMapEntry(ctx context.Context, key string, value st
 		cm.Data[key] = value
 
 		if err = s.updateObject(ctx, cm); err != nil {
-			ctrl.LoggerFrom(ctx).Error(err, "Failed to update configmap to update an entry", "namespacedName", ins, "key", key, "value", value)
-			return err
+			return fmt.Errorf("failed to update configmap %s/%s, key %s: %w", ins.Namespace, ins.Name, key, err)
```
[the remark this answers is not in the document]
Comment (yevgeny-shnaidman) [reply pkg/storage/storage.go]: fixed

File: pkg/storage/storage.go [hunk 3]
```diff
@@ -39,8 +39,7 @@ func (s *storage) CheckConfigMapEntry(ctx context.Context, key string, ins types
 func (s *storage) UpdateConfigMapEntry(ctx context.Context, key string, value string, ins types.NamespacedName) error {
 	cm, err := s.getConfigMap(ctx, ins.Namespace, ins.Name)
 	if err != nil {
-		ctrl.LoggerFrom(ctx).Error(err, "Failed to get configmap to update an entry", "namespacedName", ins, "key", key, "value", value)
-		return err
+		return fmt.Errorf("failed to get configmap %s/%s: %w", ins.Namespace, ins.Name, err)
```
[the remark this answers is not in the document]
Comment (yevgeny-shnaidman) [reply pkg/storage/storage.go]: fixed

File: pkg/storage/storage.go [hunk 4]
```diff
@@ -30,7 +30,7 @@ func NewStorage(kubeClient clients.ClientsInterface) Storage {
 func (s *storage) CheckConfigMapEntry(ctx context.Context, key string, ins types.NamespacedName) (string, error) {
 	cm, err := s.getConfigMap(ctx, ins.Namespace, ins.Name)
 	if err != nil {
-		return "", err
+		return "", fmt.Errorf("failed to get config map %s/%s: %w", ins.Namespace, ins.Name, err)
```
[the remark this answers is not in the document]
Comment (yevgeny-shnaidman) [reply pkg/storage/storage.go]: fixed

File: pkg/storage/storage.go [hunk 5]
```diff
@@ -85,11 +81,13 @@ func (s *storage) getConfigMap(ctx context.Context, namespace string, name strin
 	err := s.kubeClient.Get(ctx, dep, cm)
 
 	if apierrors.IsNotFound(err) {
-		ctrl.LoggerFrom(ctx).Error(err, "Failed to get configmap", "cmNamespace", namespace, "cmName", name)
-		return nil, err
+		return nil, fmt.Errorf("failed to find configmap %s: %w", dep, err)
```
Comment (ybettan) [inline pkg/storage/storage.go:84]: Isn't it sufficient to only check `if err != nil` in this case? Looks like `apierrors.IsNotFound(err)` knows how to handle `nil`.

**Pull request merged**